src/phoenix/trace/v1/utils.py

```diff
@@ -358,21 +357,19 @@ def _encode_io_value(
     mime_type: Optional[MimeType],
 ) -> pb.Span.IOValue:
     if mime_type is MimeType.JSON:
-        struct = Struct()
-        if io_value:
-            struct.update(json.loads(io_value))
-        return pb.Span.IOValue(json_value=struct)
-    return pb.Span.IOValue(string_value=io_value)
+        return pb.Span.IOValue(
+            value=io_value,
+            mime_type=pb.Span.IOValue.MimeType.JSON,
+        )
+    return pb.Span.IOValue(value=io_value)
 
 
 def _decode_io_value(
     pb_io_value: pb.Span.IOValue,
 ) -> Iterator[Union[str, MimeType]]:
-    if pb_io_value.WhichOneof("kind") == "json_value":
-        yield json.dumps(MessageToDict(pb_io_value.json_value))
+    yield pb_io_value.value
+    if pb_io_value.mime_type is pb.Span.IOValue.MimeType.JSON:
```

Comment:
```suggestion
    if pb_io_value.mime_type is not pb.Span.IOValue.MimeType.TEXT:
```

Maybe someday we introduce new mime types.